executor/delete.go

```diff
@@ -91,9 +92,9 @@ func (e *DeleteExec) deleteSingleTableByChunk(ctx context.Context) error {
 		break
 	}
 
-	// If tidb_batch_delete is ON and not in a transaction, we could use BatchDelete mode.
-	batchDelete := e.ctx.GetSessionVars().BatchDelete && !e.ctx.GetSessionVars().InTxn()
-	batchDMLSize := e.ctx.GetSessionVars().DMLBatchSize
+	sessVars := e.ctx.GetSessionVars()
+	// If tidb_batch_delete or enable-batch-dml is ON and not in a transaction, we could use BatchDelete mode.
```

Comment:
This comment doesn't quite match the code.
The comment like `tidb_batch_delete  && !sessVars.InTxn() ||  enable-batch-dml && !sessVars.InTxn()` or `(tidb_batch_delete  ||  enable-batch-dml) && !sessVars.InTxn()`.

Reply:
ah, yes.